lib/puppet/functions/lookup.rb

```diff
@@ -1,153 +1,128 @@
+# following keys:
+#     * `'name'` --- Same as `<NAME>` (argument 1). You can pass this as an
+#     argument or in the hash, but not both.
+#     * `'value_type'` --- Same as `<VALUE TYPE>` (argument 2).
+#     * `'merge'` --- Same as `<MERGE BEHAVIOR>` (argument 3).
+#     * `'default_value'` --- Same as `<DEFAULT VALUE>` (argument 4).
+#     * `'default_values_hash'` (hash) --- A hash of lookup keys and default
+#     values. If Puppet can't find a normal value, it will check this hash for the
+#     requested key before giving up. You can combine this with `default_value` or
+#     a lambda, which will be used if the key isn't present in this hash. Defaults
+#     to an empty hash.
+#     * `'override'` (hash) --- A hash of lookup keys and override values. Puppet
+#     will check for the requested key in the overrides hash _first;_ if found, it
+#     returns that value as the _final_ value, ignoring merge behavior. Defaults
+#     to an empty hash.
+#
+# Finally, `lookup` can take a lambda, which should accept a single parameter.
+# This is yet another way to set a default value for the lookup; if no results are
+# found, Puppet will pass the requested key to the lambda and use its result as
+# the default value.
+#
+# #### Merge Behaviors
+#
+# Puppet lookup uses a hierarchy of data sources, and a given key might have
+# values in multiple sources. By default, Puppet returns the first value it finds,
+# but it can also continue searching and merge all the values together.
+#
+# **Note:** Data sources can use the special `lookup_options` metadata key to
+# request a specific merge behavior for a key. The `lookup` function will use that
+# requested behavior unless you explicitly specify one.
+#
+# The valid merge behaviors are:
+#
+# * `'first'` --- Returns the first value found, with no merging. Puppet lookup's
+# default behavior.
+# * `'unique'` (AKA "array merge") --- Combines any number of arrays and scalar
+# values to return a merged, flattened array with all duplicate values removed.
+# The lookup will fail if any hash values are found.
+# * `'hash'` --- Combines the keys and values of any number of hashes to return a
+# merged hash. If the same key exists in multiple source hashes, Puppet will use
+# the value from the highest-priority data source; it won't recursively merge the
+# values.
+# * `'deep'` --- Combines the keys and values of any number of hashes to return a
+# merged hash. If the same key exists in multiple source hashes, Puppet will
+# recursively merge hash or array values (with duplicate values removed from
+# arrays). For conflicting scalar values, the highest-priority value will win.
+# * `{'strategy' => 'first|unique|hash'}` --- Same as the string versions of these
+# merge behaviors.
+# * `{'strategy' => 'deep', <DEEP OPTION> => <VALUE>, ...}` --- Same as `'deep'`,
+# but can adjust the merge with additional options. The `strategy` key is
+# required, and you can set any of these optional keys:
```

Comment:
What does "The `strategy` key is required" mean here? Does it mean you can't set options for the `deep` merge behavior unless you use the `strategy` hash? Or does it mean you always have to use the `strategy` hash? I'm guessing it's the former.